psl-parser/src/main/java/org/linqs/psl/parser/CommandLineLoader.java

```diff
@@ -0,0 +1,421 @@
+//delete below 4 imports? never used in both files?
+/* 
+import org.apache.log4j.BasicConfigurator;
+import org.apache.log4j.ConsoleAppender;
+import org.apache.log4j.PatternLayout;
+import org.apache.log4j.Priority;
+*/
+import org.apache.log4j.PropertyConfigurator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.UnknownHostException;
+import java.net.InetAddress;
+import java.nio.file.Paths;
+import java.util.Comparator;
+import java.util.Properties;
+import java.util.Map;
+
+/**
+ * Load the command line options into PSL Config's configuration values and 
+ * log4j configuration appropriately. This is done by instatiating the object
+ * of type CommandLine and object log of type Logger.
+ */
+public class CommandLineLoader {
+    // Command line options.
+    public static final String OPTION_HELP = "h";
+    public static final String OPTION_HELP_LONG = "help";
+    public static final String OPERATION_INFER = "i";
+    public static final String OPERATION_INFER_LONG = "infer";
+    public static final String OPERATION_LEARN = "l";
+    public static final String OPERATION_LEARN_LONG = "learn";
+
+    public static final String OPTION_DATA = "d";
+    public static final String OPTION_DATA_LONG = "data";
+    public static final String OPTION_DB_H2_PATH = "h2path";
+    public static final String OPTION_DB_POSTGRESQL_NAME = "postgres";
+    public static final String OPTION_EVAL = "e";
+    public static final String OPTION_EVAL_LONG = "eval";
+    public static final String OPTION_INT_IDS = "int";
+    public static final String OPTION_INT_IDS_LONG = "int-ids";
+    public static final String OPTION_LOG4J = "4j";
+    public static final String OPTION_LOG4J_LONG = "log4j";
+    public static final String OPTION_MODEL = "m";
+    public static final String OPTION_MODEL_LONG = "model";
+    public static final String OPTION_OUTPUT_DIR = "o";
+    public static final String OPTION_OUTPUT_DIR_LONG = "output";
+    public static final String OPTION_OUTPUT_GROUND_RULES_LONG = "groundrules";
+    public static final String OPTION_OUTPUT_SATISFACTION_LONG = "satisfaction";
+    public static final String OPTION_PROPERTIES = "D";
+    public static final String OPTION_PROPERTIES_FILE = "p";
+    public static final String OPTION_PROPERTIES_FILE_LONG = "properties";
+    public static final String OPTION_VERSION = "v";
+    public static final String OPTION_VERSION_LONG = "version";
+
+    public static final String DEFAULT_H2_DB_PATH =
+    Paths.get(System.getProperty("java.io.tmpdir"),
+    "cli_" + System.getProperty("user.name") + "@" + getHostname()).toString();
+    public static final String DEFAULT_POSTGRES_DB_NAME = "psl_cli";
+    public static final String DEFAULT_IA = MPEInference.class.getName();
+    public static final String DEFAULT_WLA = MaxLikelihoodMPE.class.getName();
+
+    public CommandLine options;
+    public Logger log;
```

Comment:
TLDR: Experiment result: The logging framework reconfigures itself on updates. Since the code works as expected, I am marking this issue as resolved. 

I tried the above experiment in expectation of 2 cases. (i) The logging framework reconfigures itself or (ii) It does not reconfigure itself. Outside this channel I had reported that the experiment results was case ii. This is wrong. I got confused with the DEBUG levels. More specifically, I was wrongly expecting log level DEBUG to print only DEBUG level logs. I assumed the INFO logs came from previous config. Instead it is from DEBUG's setting itself. So, please ignore that communication.

PF the logs of the experiments: 
Code snippet:
    public static void main(String[] args, boolean rethrow) {
        System.out.println("test3 logger");
        Logger log = LoggerFactory.getLogger(Launcher.class);
        Properties props = new Properties();
        props.setProperty("log4j.rootLogger", "DEBUG, A1");
        props.setProperty("log4j.appender.A1", "org.apache.log4j.ConsoleAppender");
        props.setProperty("log4j.appender.A1.layout", "org.apache.log4j.PatternLayout");
        props.setProperty("log4j.appender.A1.layout.ConversionPattern", "SHRBS %-4r [%t] %-5p %c %x - %m%n");
        PropertyConfigurator.configure(props);
        
        log.info("log print test logger1");
        log.debug("log print test logger2");

        props.setProperty("log4j.appender.A1.layout.ConversionPattern", "SHRESTABS %-4r [%t] %-5p %c %x - %m%n");
        props.setProperty("log4j.rootLogger", "INFO, A1");
        PropertyConfigurator.configure(props);
        log.info("log print test logger3");
        log.debug("log print test logger4");
    }

Ouput that proofs reconfiguration: 
shrestabs@[/shresta/views/psl-eriq] $ java -cp /Users/shrestabs/.m2/repository/org/linqs/psl-cli/2.2.0-SNAPSHOT/psl-cli-2.2.0-SNAPSHOT.jar org/linqs/psl/cli/Launcher
test3 logger
SHRBS 0    [main] INFO  org.linqs.psl.cli.Launcher  - log print test logger1
SHRBS 3    [main] DEBUG org.linqs.psl.cli.Launcher  - log print test logger2
SHRESTABS 3    [main] INFO  org.linqs.psl.cli.Launcher  - log print test logger3

Reply:
Good to hear that this came out as we hoped.